src/sys/mman.rs

```diff
@@ -603,3 +603,64 @@ pub fn shm_unlink<P: ?Sized + NixPath>(name: &P) -> Result<()> {
 
     Errno::result(ret).map(drop)
 }
+
+/// Creates and returns a new, or returns an existing, System V shared memory
+/// segment identifier.
+///
+/// For more information, see [`shmget(2)`].
+///
+/// [`shmget(2)`]: https://man7.org/linux/man-pages/man2/shmget.2.html
+pub fn shmget(key: key_t, size: size_t, shmflg: c_int) -> Result<c_int> {
+    Errno::result(unsafe { libc::shmget(key, size, shmflg) })
+}
+
+/// Attaches the System V shared memory segment identified by `shmid` to the
+/// address space of the calling process.
+///
+/// For more information, see [`shmat(2)`].
+///
+/// # Safety
+///
+/// `shmid` should be a valid shared memory identifier and
+/// `shmaddr` must meet the requirements described in the [`shmat(2)`] man page.
+///
+/// [`shmat(2)`]: https://man7.org/linux/man-pages/man2/shmat.2.html
+pub unsafe fn shmat(
+    shmid: c_int,
+    shmaddr: *const c_void,
+    shmflg: c_int,
```

Comment:
Again: `shmflg`, `shmid` should be wrapped in safe abstractions. Reading the man pages it even seems like shmat and shmget take different sets of flags. So two different enums probably?